### DISCUSSION TITLE
City environment

### DISCUSSION THREAD

**Discussion by Flautdn** [microsoft/AirSim#4180] [Support Q&A]
I got a built in Blocks environment from airsim installation in Unreal/environments folder which I used for initial Unreal project work.

Now I want City environment folder without binaries(sources only) like built in blocks environment airsim source folder gives me.

Where can I get City environment folder so that I can edit the simMode option and make changes that the environment will become compatible with.

Thanks

**Comment by zimmy87** [comment]
Unfortunately, the City Environment is a proprietary environment and we don't have the appropriate license to distribute the source code for it.

**Reply by Flautdn** [reply]
There are world NED co-ordinate system and vehicle's NED co-ordinate system.
When an api assigns desired distance or velocity in vehicle's NED co-ordinate system how to interpret in
reference to the world co-ordinate system.

**Reply by zimmy87** [reply]
This [doc](https://microsoft.github.io/AirSim/apis/#coordinate-system) should help answer your question. AirSim's units are in meters whereas Unreal's units are centimeters, so you'll need to multiply or divide by 100 when converting units. When the simulation starts, we set the origin of the world to where the first drone spawns, so if you're obtaining coordinates of objects inside the simulation, you'll need to offset by the position of the PlayerStart to convert to coordinates in the editor.

**Comment by Flautdn** [comment]
So we only get blocks with the source code in the listed airsim releases page?

Thank you.

**Reply by zimmy87** [reply]
Correct, some of the environments can be found in the Unreal Marketplace, but Blocks is the only one we distribute source for in this repository.

**Reply by Flautdn** [reply]
Is it possible to setup the vehicles in the simualtion in a way with some having the apienablecontrol=true and others with apienbalecontrol=false ?
I am unable to move the vehicles with apienablecontrol=true when simulating.

Thanks.

**Reply by zimmy87** [reply]
There's an optional vehicle_name parameter for the `client.enableApiControl(is_enabled, vehicle_name)` API. It defaults to an empty string which should enable/disable API control for the default drone. If you're spawning multiple vehicles or have named the vehicle to something other than the default name, you may need to call `client.enableApiControl()` for each vehicle.

**Reply by Flautdn** [reply]
Okay Thanks will do that

**Comment by Flautdn** [comment]
Okay the blocks folder structure is different from what I see in the unreal marketplace or few others like turbosquid.com or cgitrader.com 
When I use blocks environment I am able to simulate car&drone in blocks with airsim seamelessly.

In the environment downloaded from turbosquid, file structure is different and the environment is not getting loaded like in the case of blocks.

What are the changes to be made if I take an environment from the unreal market place so that  I can compile and use it just like blocks for airsim simulation purpose.


Thank you.

**Reply by zimmy87** [reply]
You should be able to follow the steps [here](https://microsoft.github.io/AirSim/unreal_custenv/) to configure an environment for AirSim. There's both some project settings that need to be set and the AirSim plugin which needs to be added to the project.

**Comment by Flautdn** [comment]
I am trying send image requests to my drone.

Do the drone have more than one camera by default without configuring in the settings. json file.
Also how can I get the list of parameters that a camera sensor can have like
focal length (also how do I know default focal length value for a camera sensor)
setting the resolution parameters(height width)

**Reply by zimmy87** [reply]
see my comment [here](https://github.com/microsoft/AirSim/discussions/4212#discussioncomment-1811479)

**Comment by Flautdn** [comment]
how to add the camera sensor and it's format in settings.json

**Reply by zimmy87** [reply]
see the doc on camera settings [here](https://microsoft.github.io/AirSim/settings/#camera-settings)

**Comment by Flautdn** [comment]
Hi there is a way of obtaining depth from drone cameras with the api Imagerequest() with Image Types available are
DepthPlanar
DepthPerspective

How do they differ and and how to get focal length for cameras and how to modify focal length 

Is fov constant along the two camera axes (i see default fov = 90)

Thanks.

**Reply by zimmy87** [reply]
see doc on DepthPlanar & DepthPerspective [here](https://microsoft.github.io/AirSim/image_apis/#depthplanar-and-depthperspective)

**Comment by Flautdn** [comment]
Hi
      I am using the camera and got the disparity map. How can I get the
camera f value or other intrinsic camera parameters for getting real world
co-ordinates from image co-ordinates.

Thanks.

Regards.

**Reply by zimmy87** [reply]
See my comment [here](https://github.com/microsoft/AirSim/discussions/4212#discussioncomment-1811479). Answers to a lot of your questions can be found by searching in the docs @ https://microsoft.github.io/AirSim/. You should be able to get help quicker that way rather than waiting for an answer from me or someone else in the discussions.
